mailbox/backup/src/test/java/org/apache/james/mailbox/backup/MailBoxAnnotationsExtraFieldTest.java

```diff
@@ -0,0 +1,283 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.backup;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ContiguousSet;
+import com.google.common.collect.DiscreteDomain;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Range;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class MailBoxAnnotationsExtraFieldTest {
+
+    private String annotation1Serialized = MailboxMessageFixture.ANNOTATION_1.getKey().asString() + ":" + MailboxMessageFixture.ANNOTATION_1.getValue().orElse("");
+
+    private int sizeAnnotation1 = 1 + MailboxMessageFixture.ANNOTATION_1.getKey().asString().length() + MailboxMessageFixture.ANNOTATION_1.getValue().orElse("").length();
+    private int sizeAnnotation2 = 1 + MailboxMessageFixture.ANNOTATION_2.getKey().asString().length() + MailboxMessageFixture.ANNOTATION_2.getValue().orElse("").length();
+    private int sizeSeparator = 1;
+
+    private List<MailboxAnnotation> oneThousandTimes =   ContiguousSet.create(Range.closed(0, 999), DiscreteDomain.integers()).stream()
+            .map(i -> MailboxMessageFixture.ANNOTATION_1).collect(Collectors.toList());
+
+    private String bufferContentForOneThousand = oneThousandTimes.stream().reduce("", (acc, e) ->  {
```

Comment:
IMO we can make this easier to read using a `map` step and then using the joining collector, no?